### PR TITLE
build: Mockery v1.1.1

### PR DESCRIPTION
**Pull request by alexec**
Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to USERS.md.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -233,18 +233,21 @@ dist/argoexec-%: $(ARGOEXEC_PKGS)
...
-$(HOME)/go/bin/mockery:
-	$(call backup_go_mod)
-	go get github.com/vektra/mockery/.../
-	$(call restore_go_mod)
+$(GOPATH)/bin/mockery:
+	./hack/recurl.sh dist/mockery.tar.gz https://github.com/vektra/mockery/releases/download/v1.1.1/mockery_1.1.1_$(shell uname -s)_$(shell uname -m).tar.gz
+	tar zxvf dist/mockery.tar.gz mockery
+	chmod +x mockery
+	mkdir -p $(GOPATH)/bin
+	mv mockery $(GOPATH)/bin/mockery
```
Comment (simster7) [inline Makefile:246]: 1. Manually overwriting binaries in `go/bin` might break some existing users' flow with `go get`. The download is 2 MB... perhaps we should keep in in the `dist` folder?

1. What happens if the file already exists?

File: Makefile [hunk 2]
```diff
@@ -277,13 +280,15 @@ manifests:
...
+$(GOPATH)/bin/golangci-lint:
 	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b `go env GOPATH`/bin v1.23.8
 
 .PHONY: lint
-lint: server/static/files.go $(HOME)/go/bin/golangci-lint
+lint: server/static/files.go $(GOPATH)/bin/golangci-lint
 	# Tidy Go modules
 	go mod tidy
+	# Build to find errors that will prevent linting.
+	go build ./...
```
Comment (simster7) [inline Makefile:291]: Is there a reason you added this?

I thought that the linter itself detects build errors. Seems like this line would just make `make lint` take longer

File: Makefile (hunk 1 above)
Comment (alexec) [reply Makefile:246]: if the file already exists, nothing happens, we want to install it permanently, we could go back to `go get`?

Comment (alexec) [reply Makefile:246]: reason to install permanently = reliability + speed

File: Makefile (hunk 2 above)
Comment (alexec) [reply Makefile:291]: I'll revert this line- it's not needed.

File: Makefile (hunk 1 above)
Comment (simster7) [reply Makefile:246]: > if the file already exists, nothing happens

This is good enough, then. IMO

File: Makefile (hunk 2 above)
Comment (alexec) [reply Makefile:291]: Ops - I don't want to revert - there's a class of builds errors that break the linter - when SSA cannot be created. This prevents this. It does assume we want to build it mind you - not always the case.

**Pull request merged**